scripts/windows/msys2_prepare.bat

```diff
@@ -0,0 +1,59 @@
+@echo on
+@rem Script to prepare msys2 environment for builds.
+@rem Required vars:
+@rem    ERTS_VERSION
+@rem    OTP_VERSION
+@rem    OTP_PATH
+@rem    MSYS2_ROOT
+@rem    PLATFORM
+
+SETLOCAL ENABLEEXTENSIONS
+
+rem Set required vars defaults
+IF "%ERTS_VERSION%"=="" SET "ERTS_VERSION=9.3"
+IF "%OTP_VERSION%"=="" SET "OTP_VERSION=20.3"
+IF "%OTP_PATH%"=="" SET "OTP_PATH=C:\Program Files\erl"
+IF "%MSYS2_ROOT%"=="" SET "MSYS2_ROOT=C:\msys64"
+IF "%PLATFORM%"=="" SET "PLATFORM=x64"
+
+@echo Current time: %time%
+rem Set the paths appropriately
+
+call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\\vcvarsall.bat" %PLATFORM%
+@echo on
+SET PATH=%MSYS2_ROOT%\mingw64\bin;%MSYS2_ROOT%\usr\bin;%PATH%
+
+@echo Current time: %time%
+rem Upgrade the MSYS2 platform
+
+bash -lc "pacman --noconfirm --needed -Sy pacman"
+bash -lc "pacman --noconfirm -Su"
+
+@echo Current time: %time%
+rem Install required tools
+
+bash -lc "pacman --noconfirm --needed -S base-devel cmake curl gcc gcc-fortran git make patch"
+bash -lc "pacman --noconfirm --needed -S mingw-w64-x86_64-SDL mingw-w64-x86_64-libsodium mingw-w64-x86_64-ntldd-git mingw-w64-x86_64-yasm"
+bash -lc "pacman --noconfirm --needed -S mingw-w64-x86_64-gcc mingw-w64-x86_64-gcc-ada mingw-w64-x86_64-gcc-fortran mingw-w64-x86_64-gcc-objc"
```

Comment:
It would be nice to test whether `mingw-w64-x86_64-gcc-ada mingw-w64-x86_64-gcc-fortran mingw-w64-x86_64-gcc-objc` are needed.

Reply:
They are needed, because there are some cross-dependencies which are hard to untangle via a single script.